core/store/src/main/java/org/locationtech/geowave/core/store/api/DataStore.java

```diff
@@ -131,25 +132,42 @@ void addIndex(
 			Index... indices );
 
 	/**
-	 * remove statistics, across types
+	 * This method removes an index from the index store. It is also
+	 * responsible for removing the index's mapping, data, and stats 
+	 * for each adapter/type that was associated with the index. If this
+	 * is the last index associated with any of the existing adapters,
+	 * the method will throw an exception.
 	 * 
-	 * if this is the last index for a type need to remove type first or throw exception
 	 * @param indexName
 	 */
 	void removeIndex(
-			String indexName );
+			String indexName )
+			throws AdapterException;
 
-	void removeIndex(
+	/**
+	 * This method removes an index mapping with a given adapter/type. It is 
+	 * also data, and stats for the adapter that was associated with the index.
+	 * If this is the last index associated with type the method will throw 
+	 * an exception.
+	 * 
+	 * @param typeName
+	 * @param indexName
+	 */
+	void removeIndexMapping(
```

Comment:
AdapterIndexMapping is an internal thing, so to me "mapping" makes less sense on the API than simply removeIndex

Reply:
Naming is reverted